code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/manager/ServiceResourceManager.java

```diff
@@ -0,0 +1,54 @@
+package io.cattle.platform.servicediscovery.api.manager;
+
+import io.cattle.platform.api.resource.jooq.AbstractJooqResourceManager;
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.servicediscovery.api.constants.ServiceDiscoveryConstants;
+import io.cattle.platform.servicediscovery.api.dao.ServiceExposeMapDao;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+public class ServiceResourceManager extends AbstractJooqResourceManager {
+    
+    @Inject
+    ServiceExposeMapDao exposeMapDao;
+
+    @Inject
+    ObjectManager objectManager;
+
+    @Override
+    public String[] getTypes() {
+        return new String[0];
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[] { Service.class };
+    }
+
+    @Override
+    protected Object updateInternal(String type, String id, Object obj, ApiRequest request) {
```

Comment:
@ibuildthecloud if we decide to assume that the desired scale is the max scale, then we won't need this method - it was to update the scale for all the participants on a single service scaleup